core/src/main/scala/kafka/server/MetadataCache.scala

```diff
@@ -184,6 +185,7 @@ private[server] class MetadataCache(brokerId: Int) extends Logging {
           removePartitionInfo(tp.topic, tp.partition)
           stateChangeLogger.trace(s"Broker $brokerId deleted partition $tp from metadata cache in response to UpdateMetadata " +
             s"request sent by controller $controllerId epoch $controllerEpoch with correlation id $correlationId")
+          groupCoordinator.groupManager.cleanupGroupMetadata(tp)
```

Comment:
One minor suggestion: It's a bit unfortunate that `MetadataCache` now depends on `GroupCoordinator`, right? I'm wondering if it would be reasonable to let `updateCache` return the set of the removed partitions, which could then be passed to `cleanupGroupMetadata` from `KafkaApis` in the UpdateMetadata request handler?

One other nit: we usually do not make direct references to `GroupMetadataManager` outside of `GroupCoordinator`, preferring delegation instead (see `GroupCoordinator.handleGroupEmigration` for example).